src/components/MultiSelect.tsx

```diff
@@ -45,17 +43,21 @@ export type Props = {
   localeNamespace?: LocaleNamespace
   menuPortalTarget?: HTMLElement
   onFocus?: () => void
-  options: Options<number | string> | ((lan: Language) => Options<number | string>)
+  options:
+    | Options<number>
+    | Options<string>
+    | ((lan: Language) => Options<string>)
+    | ((lan: Language) => Options<number>)
   overwriteIsMobileTouch?: boolean // For testing purposes only
   priority?: Priority // Show on top of select options
   readOnly?: boolean
   style?: Partial<ComponentTheme['select']>
-  value: Value
   onChange: (value: InternalOption[]) => void
   onBlur?: (value: InternalOption[]) => void
+  value: T[]
 }
 
-export const MultiSelect = (props: Props) => {
+export const MultiSelect = (props: Props<string> | Props<number>) => {
```

Comment:
The generic would rather need to be set as generic here, no?
`export const MultiSelect = <T>(props: Props<T>) => { ...`